### PR TITLE
feat(metric_alerts): Allow trigger thresholds to be float

### PR DESCRIPTION
**Pull request by wedamija**
This updates the data model to convert the threshold fields from int -> float.

The data migration should be fine since this table is small/not on a hot path. I tested having code
where the model still has `models.IntegerField` defined, but the database has `double`, and
everything continued to work as usual - fetches and updates all worked fine and didn't cause type
conversion errors.

### CODE REVIEW

**Comment by wedamija** [comment]
Generated DDL
```
BEGIN;
--
-- Alter field alert_threshold on alertruletrigger
--
ALTER TABLE "sentry_alertruletrigger" ALTER COLUMN "alert_threshold" TYPE numeric(30, 10) USING "alert_threshold"::numeric(30, 10);
--
-- Alter field resolve_threshold on alertruletrigger
--
ALTER TABLE "sentry_alertruletrigger" ALTER COLUMN "resolve_threshold" TYPE numeric(30, 10) USING "resolve_threshold"::numeric(30, 10);
COMMIT;
```

cc @sydjryan, I doubt this will cause you any problems but we're changing a column data type here.

File: src/sentry/migrations/0081_alert_rules_threshold_float.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            model_name='alertruletrigger',
+            name='alert_threshold',
+            field=models.FloatField(),
+        ),
+        migrations.AlterField(
+            model_name='alertruletrigger',
+            name='resolve_threshold',
+            field=models.FloatField(null=True),
+        ),
+        migrations.AlterField(
```
Comment (wedamija) [inline src/sentry/migrations/0081_alert_rules_threshold_float.py:43]: This is just an unrelated change caused by someone modifying the state on this model. It doesn't cause any db changes.

**Comment by sydjryan** [comment]
this is fine with me, we're not pulling this table into bigquery yet. thanks for the heads up! 👍

**Comment by sydjryan** [comment]
PS including the DDL is huuuugely helpful, much clearer for me to see what's going on ✨

File: src/sentry/incidents/models.py [hunk 2]
```diff
@@ -469,8 +469,8 @@ class AlertRuleTrigger(Model):
     alert_rule = FlexibleForeignKey("sentry.AlertRule")
     label = models.TextField()
     threshold_type = models.SmallIntegerField()
-    alert_threshold = models.IntegerField()
-    resolve_threshold = models.IntegerField(null=True)
+    alert_threshold = models.FloatField()
+    resolve_threshold = models.FloatField(null=True)
```
Comment (markstory) [inline src/sentry/incidents/models.py:473]: How precise would these values be? Will you run into float precision problems when you read these values back out or compare these values with other data? A DecimalField could help avoid those problems.

Comment (wedamija) [reply src/sentry/incidents/models.py:473]: I actually had these as `DecimalField` to being with but wasn't sure how useful it was considering we're passing floats from the frontend anyway. I suppose we could start passing them as strings.

I'm not sure we're going to have extremely precise values here - we won't be doing equality comparisons, and the values will likely be 2-4 decimal places at most. If you think it's worthwhile I'm good to change it though.

Comment (EvanPurkhiser) [reply src/sentry/incidents/models.py:473]: Yeah we won't be doing math on these or anything like that, so I think we should be safe to use a float here.

**Review by evanpurkhiser — APPROVED** [review]

**Review by markstory — APPROVED** [review]
DDL changes look good to me. These are small tables so locks shouldn't cause problems.
